### ISSUE TITLE
[BUG] Microsoft.Azure.Management.SignalR.Models.SignalRFeature: inconsistency with resource manifest/template because of static member Flag

### ISSUE THREAD

**Issue by karok2m** [Azure/azure-sdk-for-net#8806]
**Describe the bug**
In `Microsoft.Azure.Management.SignalR.Models.SignalRFeature` class, there is no way to set value of `Flag` member to `"ServiceMode"`. And it being a `static string` causes issues when object of type `Microsoft.Azure.Management.SignalR.Models.SignalRResource` is created (through `signalRManagementClient.SignalR.CreateOrUpdateAsync(...)`), that `Flag`s of both returned `Features` are set to `"EnableConnectivityLogs"`. This is in conflict with manifest/template of the created resource in Azure Portal, which shows the following `Features`:

```json
"features": [
    {
        "flag": "ServiceMode",
        "value": "Default",
        "properties": {}
    },
    {
        "flag": "EnableConnectivityLogs",
        "value": "True",
        "properties": {}
    }
]
```

**Microsoft.Azure.Management.SignalR** NuGet Version: **1.0.1**

***Exception or Stack Trace***
NA

**To Reproduce**
Create instance of SiglaR through **.Net Core SDK**. Use code snippet provided bellow.

***Code Snippet***
```csharp
var signalRCreateParameters = new SignalRCreateParameters() {
    Location = resourceGroup.RegionName,
    Tags = tags,

    Sku = new ResourceSku {
        Name = "Standard_S1",
        Tier = "Standard",
        Capacity = 1,
    },
    Properties = new SignalRCreateOrUpdateProperties {
        HostNamePrefix = signalRName,
        Features = new List<SignalRFeature> {
            new SignalRFeature {
                Value = "Default"
            }
        }
    }
};

signalRCreateParameters.Validate();

var signalR = await _signalRManagementClient
    .SignalR
    .CreateOrUpdateAsync(
        resourceGroup.Name,
        signalRName,
        signalRCreateParameters,
        cancellationToken
);
```

**Expected behavior**
* It is possible to set `Flag` member of `Microsoft.Azure.Management.SignalR.Models.SignalRFeature` class. For example to `"ServiceMode"` or `"EnableConnectivityLogs"`.
* When `Microsoft.Azure.Management.SignalR.Models.SignalRResource` is created, it can have `Features` with different `Flag`s. In my case:
  * one whose `Flag` is equal to `"ServiceMode"`, and `Value` to `"Default"`
  * one whose `Flag` is equal to `"EnableConnectivityLogs"`, and `Value` to `"True"`

**Setup (please complete the following information):**
 - OS: Windows 10 Enterprise x64 (1909)
 - IDE : Visual Studio Enterprise 2019, Version 16.3.9
 - **Microsoft.Azure.Management.SignalR** NuGet Version: **1.0.1**

**Information Checklist**
Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [x] Bug Description Added
- [x] Repro Steps Added
- [x] Setup information Added

**Comment by isra-fel** [comment]
Sorry for late response. @sffamily, @chenkennt could you please share some insights? Thank you

**Comment by chenkennt** [comment]
@juniwang please help take a look.

**Comment by juniwang** [comment]
Sorry for late response. We have a [PR](https://github.com/Azure/azure-rest-api-specs/pull/8540) under reviewing, a preview of the .net SDK change can be found at https://github.com/AzureSDKAutomation/azure-sdk-for-net/pull/585/files#diff-32198cbb0ddba3a6b70e55a0eff49202L42. After this PR merged, it will be fixed in the new version of .NET SDK.

In older versions of SDK, since `ServiceMode` is the only public `FeatureFlag`, so the auto-generated SDK used a static `flag`.  It's a by-design behavior of `autorest` which we use to generate Azure management SDKs. In the upcoming new version, another `FeatureFlag` introduced, then both `flag` and `value` are required to set.

**Comment by karok2m** [comment]
@juniwang, should this issue be closed as [#9377](https://github.com/Azure/azure-rest-api-specs/pull/9377) is merged ?

**Comment by juniwang** [comment]
Yes. it's resolved
